Fix maybe-uninitialized warnings in posix.cc

GCC 10.2 emits some new warnings:

```
/mnt/hdf/vchoi/work/MY_GIT_CLONE/HERMES3/hermes/adapter/posix/posix.cc: In function 'ssize_t pread(int, void, size_t, off_t)':
/mnt/hdf/vchoi/work/MY_GIT_CLONE/HERMES3/hermes/adapter/posix/posix.cc:555:14: error: 'ret' may be used uninitialized in this function [-Werror=maybe-uninitialized]
555 | return (ret);
| ^
/mnt/hdf/vchoi/work/MY_GIT_CLONE/HERMES3/hermes/adapter/posix/posix.cc: In function 'ssize_t pread64(int, void, size_t, off64_t)':
/mnt/hdf/vchoi/work/MY_GIT_CLONE/HERMES3/hermes/adapter/posix/posix.cc:600:14: error: 'ret' may be used uninitialized in this function [-Werror=maybe-uninitialized]
600 | return (ret);
| ^
/mnt/hdf/vchoi/work/MY_GIT_CLONE/HERMES3/hermes/adapter/posix/posix.cc: In function 'ssize_t pwrite(int, const void, size_t, off_t)':
/mnt/hdf/vchoi/work/MY_GIT_CLONE/HERMES3/hermes/adapter/posix/posix.cc:578:14: error: 'ret' may be used uninitialized in this function [-Werror=maybe-uninitialized]
578 | return (ret);
| ^
/mnt/hdf/vchoi/work/MY_GIT_CLONE/HERMES3/hermes/adapter/posix/posix.cc: In function 'ssize_t pwrite64(int, const void, size_t, off64_t)':
/mnt/hdf/vchoi/work/MY_GIT_CLONE/HERMES3/hermes/adapter/posix/posix.cc:623:14: error: 'ret' may be used uninitialized in this function [-Werror=maybe-uninitialized]
623 | return (ret);
| ^
```

--- a/adapter/mpiio/mpiio.cc
+++ b/adapter/mpiio/mpiio.cc
@@ -501,7 +501,7 @@ int HERMES_DECL(MPI_Wait)(MPI_Request *req, MPI_Status *status) {
 }
 
 int HERMES_DECL(MPI_Waitall)(int count, MPI_Request *req, MPI_Status *status) {
-  int ret;
+  int ret = 0;
   for (int i = 0; i < count; i++) {
     auto sub_ret = MPI_Wait(&req[i], &status[i]);
     if (sub_ret != MPI_SUCCESS) {

--- a/adapter/posix/posix.cc
+++ b/adapter/posix/posix.cc
@@ -534,7 +534,7 @@ ssize_t HERMES_DECL(write)(int fd, const void *buf, size_t count) {
 }
 
 ssize_t HERMES_DECL(pread)(int fd, void *buf, size_t count, off_t offset) {
-  size_t ret;
+  size_t ret = 0;
   if (hermes::adapter::IsTracked(fd)) {
     auto mdm = hermes::adapter::Singleton<MetadataManager>::GetInstance();
     auto existing = mdm->Find(fd);
@@ -557,7 +557,7 @@ ssize_t HERMES_DECL(pread)(int fd, void *buf, size_t count, off_t offset) {
 
 ssize_t HERMES_DECL(pwrite)(int fd, const void *buf, size_t count,
                             off_t offset) {
-  size_t ret;
+  size_t ret = 0;
   if (hermes::adapter::IsTracked(fd)) {
     auto mdm = hermes::adapter::Singleton<MetadataManager>::GetInstance();
     auto existing = mdm->Find(fd);
@@ -579,7 +579,7 @@ ssize_t HERMES_DECL(pwrite)(int fd, const void *buf, size_t count,
 }
 
 ssize_t HERMES_DECL(pread64)(int fd, void *buf, size_t count, off64_t offset) {
-  size_t ret;
+  size_t ret = 0;
   if (hermes::adapter::IsTracked(fd)) {
     auto mdm = hermes::adapter::Singleton<MetadataManager>::GetInstance();
     auto existing = mdm->Find(fd);
@@ -602,7 +602,7 @@ ssize_t HERMES_DECL(pread64)(int fd, void *buf, size_t count, off64_t offset) {
 
 ssize_t HERMES_DECL(pwrite64)(int fd, const void *buf, size_t count,
                               off64_t offset) {
-  size_t ret;
+  size_t ret = 0;
   if (hermes::adapter::IsTracked(fd)) {
     auto mdm = hermes::adapter::Singleton<MetadataManager>::GetInstance();
     auto existing = mdm->Find(fd);

